### DISCUSSION TITLE
Display Voltage (Vb & Vu)

### DISCUSSION THREAD

**Discussion by VK2SOT** [dl9rdz/rdz_ttgo_sonde#455] [Q&A]
Wish to confirm the Vb = battery voltage & Vu = USB voltage? Also that the Vu is display when unplugged?

![20240831_101149](https://github.com/user-attachments/assets/d1704517-eb6f-4e1b-895a-13c2ed73f3e5)

**Comment by hr-ru** [comment]
Yes.
First letter: B=>Battery present (not charging), C=>Charging, U=>USB, no battery
Vb is battery voltage (always shown, even if there is no battery, but then it should be (close to) 0V)
Vu is USB voltage (only shown when USB connected, i.e. not shown when unplugged)

(Possibly the behavior is a bit different, depending if you are using T-Beam 1.2/AXP2101 and or T-Beam 1.0/1.1/AXP192. The AXP192 might show some Vu value even when unplugged)

**Reply by hr-ru** [reply]
On a side note: I just double-checked this with my older T-Beam.
There was an issue with USB voltage and the AXP192. It was correctly showning on the M5stack, but not on the T-Beam 1.0 (always showed 0...). This will be fixed in the next version.

**Reply by VK2SOT** [reply]
Thanks for that!

It’s a T-Beam-AXP2101-V1.2 with the battery present and USB unplugged, with devel20240521.

**Reply by hr-ru** [reply]
Thanks for this information. I did take a closer look:

- So what the software does is just read the ADC value from the AXP2101 and (usually) show that.
- If I unplug the USB cable here, typically I get very strange values, and if the ADC reads a value > 7 V, the software stops showing the USB voltage. So what you keep seeing is the last measurement before you unplugged the USB cable...
- However, it does not clear a value already on the screen, so in fact it just stops updating that part of the screen. This is somewhat not ideal and should be changed in the future...
